dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3803,26 +3820,196 @@ Sedp::TypeLookupRequestReader::process_get_types_request(const XTypes::TypeLooku
+}
+
+DDS::ReturnCode_t
+Sedp::TypeLookupReplyReader::process_get_dependencies_reply(const DCPS::ReceivedDataSample& sample,
+                                                            const XTypes::TypeLookup_Reply& reply,
+                                                            const DCPS::SequenceNumber& seq_num,
+                                                            bool is_discovery_protected)
+{
+  if (reply._cxx_return.getTypeDependencies.result.continuation_point.length() != 0 &&
+      reply._cxx_return.getTypeDependencies.result.dependent_typeids.length() == 0) {
+    ACE_ERROR((LM_WARNING, ACE_TEXT("(%P|%t) ERROR: Sedp::TypeLookupReplyReader::process_get_dependencies_reply - ")
+      ACE_TEXT("received reply with no data\n")));
+    return DDS::RETCODE_NO_DATA;
+  }
+
+  const XTypes::TypeLookup_getTypeDependencies_Out& data = reply._cxx_return.getTypeDependencies.result;
+  const DCPS::RepoId remote_id = sample.header_.publication_id_;
+
+  const XTypes::TypeIdentifier remote_ti = sedp_.orig_seq_numbers_[seq_num].type_id;
+  const GuidPrefixWrapper guid_pref(remote_id.guidPrefix);
+  XTypes::TypeIdentifierSeq& deps = dependencies_[guid_pref][remote_ti].second;
+  for (size_t i = 0; i < data.dependent_typeids.length(); ++i) {
+    const XTypes::TypeIdentifier& ti = data.dependent_typeids[i].type_id;
+    // Optimization - only store TypeIdentifiers for which TypeObjects haven't
+    // already been in the type objects cache yet
+    if (!sedp_.type_lookup_service_->type_object_in_cache(ti)) {
+      deps.append(ti);
+    }
+  }
+  dependencies_[guid_pref][remote_ti].first = data.continuation_point;
+
+  // Update internal data
+  sedp_.orig_seq_numbers_.insert(std::make_pair(++sedp_.type_lookup_service_sequence_number_,
+                                                sedp_.orig_seq_numbers_[seq_num]));
+  sedp_.orig_seq_numbers_.erase(seq_num);
+
+  if (data.continuation_point.length() == 0) { // Get all type objects
+    deps.append(remote_ti);
+    if (!sedp_.send_type_lookup_request(deps, remote_id, is_discovery_protected, true)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: Sedp::TypeLookupReplyReader::process_get_dependencies_reply - ")
+        ACE_TEXT("failed to send getTypes request\n")));
+      return DDS::RETCODE_ERROR;
+    }
+  } else { // Get more dependencies
+    XTypes::TypeIdentifierSeq type_ids;
+    type_ids.append(remote_ti);
+    if (!sedp_.send_type_lookup_request(type_ids, remote_id, is_discovery_protected, false)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: Sedp::TypeLookupReplyReader::process_get_dependencies_reply - ")
+        ACE_TEXT("failed to send getTypeDependencies request\n")));
+      return DDS::RETCODE_ERROR;
+    }
+  }
+
   return DDS::RETCODE_OK;
 }
 
+void Sedp::cleanup_type_lookup_data(const XTypes::TypeIdentifier& ti)
+{
+#ifdef OPENDDS_SECURITY
+  type_lookup_reply_secure_reader_->cleanup(ti);
+#else
+  type_lookup_reply_reader_->cleanup(ti);
```

Comment:
`type_lookup_reply_reader_` can also be used in a security-enabled build

Reply:
This has been updated.